index.bs

```diff
@@ -1,16 +1,130 @@
 <pre class='metadata'>
-Title: Your Spec Title
-Shortname: your-spec
+Title: Conversion Measurement
+Shortname: conversion-measurement
 Level: 1
-Status: ED
-Group: WGNAMEORWHATEVER
-URL: http://example.com/url-this-spec-will-live-at
-Editor: Your Name, Your Company http://example.com/your-company, your-email@example.com, http://example.com/your-personal-website
-Abstract: A short description of your spec, one or two sentences.
+Status: w3c/ED
+Group: TODO
+URL: http://csharrison.github.io/conversion-measurement-api
+Editor: Charlie Harrison, Google Inc. https://google.com, csharrison@chromium.org
+Abstract: A new API to measure and attribute cross-site conversions.
 </pre>
 
 Introduction {#intro}
 =====================
 
-Introduction here.
+<em>This section is non-normative</em>
 
+This specification describes how web browsers can provide a mechanism to the
+web that allows measuring and attributing conversions (e.g. purchases) to ads
+a user interacted with on another site. This mechanism should remove the need
+for cross site identifiers like third party cookies.
+
+## Overview ## {#overview}
+
+Anchor tags with `impressiondata` and `conversiondestination` attributes are
+classified as impression tags. When impression tags are clicked, and the
+resulting navigation commits in a document matching the `conversiondestination`,
+then the impression is stored in UA storage.
+
+At a later point, the `conversiondestination` site may fire an HTTP request to
+trigger conversion registration, which matches up with any previously
+stored impressions. If matching impressions exist, they are scheduled to be
+reported at a later time, possibly multiple days in the future.
+
+Reports are sent to reporting endpoints that are configured in impression tags
+and conversion registration requests.
+
+# Fetch monkeypatches # {#fetch-monkeypatches}
+TODO: Patch into fetch for cancelling requests redirected to the .well-known
+conversion domain.
+
+# HTML monkeypatches # {#html-monkeypatches}
+
+Rewrite the anchor element to accept the following attributes:
+
+<pre class="idl">
+partial interface HTMLAnchorElement {
+    [CEReactions=NotNeeded, Reflect] attribute DOMString conversiondestination;
+    [CEReactions=NotNeeded, Reflect] attribute DOMString impressiondata;
+    [CEReactions=NotNeeded, Reflect] attribute DOMString reportingdomain;
+    [CEReactions=NotNeeded, Reflect] attribute unsigned long long impressionexpiry;
+};
+</pre>
+
+TODO: Need monkey patches passing impression data in navigation, and a mechanism
+for validating the resulting document matches the conversiondestination.
+
+# Structures # {#structures}
+
+<h3 dfn-type=dfn>Impression</h3>
+
+An impression is a [=struct=] with the following items:
+
+<ul dfn-for="impression">
+* <dfn>impression source</dfn>: An [=url/origin=].
+* <dfn>impression data</dfn>: A {{DOMString}}.
```

Comment:
For fields in structures, we usually use the data types in Infra, like https://infra.spec.whatwg.org/#strings, instead of WebIDL data types like `DOMString`. Then when it's not obvious, use prose to describe how the notional data type from Infra is converted to the programming type in WebIDL.